draft-ietf-quic-multipath.md

```diff
@@ -1437,36 +1470,48 @@ The sequence number assigned to the connection ID by the sender on the path
 specified by Path Identifier, encoded as a variable-length integer. 
 
 
-## MAX_PATHS frames {#max-paths-frame}
+## MAX_CLIENT_PATHS and MAX_SERVER_PATHS frames {#max-paths-frame}
 
-A MAX_PATHS frame (type=0x15228c0b) informs the peer of the cumulative number of paths 
-it is permitted to open. 
+The MAX_CLIENT_PATHS and MAX_SERVER_PATHS frame (type=0x15228c0b) informs the peer of the cumulative
+number of client initiated and server initiated paths for which it MAY publish
+MP_NEW_CONNECTION_ID frames. 
 
 MAX_PATHS frames are formatted as shown in {{fig-max-paths-frame-format}}.
 
 ~~~
-MAX_PATHS Frame {
-  Type (i) = 0x15228c0b,
-  Maximum Paths (i),
+MAX_CLIENT_PATHS Frame {
+  Type (i) = 0x15228c0c,
+  Maximum Path Identifier (i),
+}
+
+MAX_SERVER_PATHS Frame {
+  Type (i) = 0x15228c0d,
+  Maximum Path Identifier (i),
 }
+
 ~~~
 {: #fig-max-paths-frame-format title="MAX_PATHS Frame Format"}
 
-MAX_PATHS frames contain the following field:
+MAX_CLIENT_PATHS and MAX_SERVER_PATHS frames contain the following field:
 
 Maximum Path Identifier:
-: A count of the cumulative number of path that can be opened 
-over the lifetime of the connection. This value cannot exceed 2^32-1, as it is not 
-possible to encode Path IDs larger than 2^32-1. Receipt of a frame that permits 
-opening of a path with Path Identifier larger than this limit MUST be treated 
-as a connection error of type FRAME_ENCODING_ERROR.
-
-Loss or reordering can cause an endpoint to receive a MAX_PATHS frame with 
-a lower path limit than was previously received. MAX_PATHS frames that 
+: A count of the cumulative number of path that can be opened
+over the lifetime of the connection. This value cannot exceed 2^32-1, as it is not
+possible to encode Path IDs larger than 2^32-1. Receipt of a frame that permits
+opening of a path with Path Identifier larger than this limit MUST be treated
+as a connection error of type FRAME_ENCODING_ERROR. This MUST be
+an even value for client initiated paths, and an odd value for
+server initiated path.
+
+Loss or reordering can cause an endpoint to receive a MAX_CLIENT_PATHS or MAX_SERVER_PATHS frame with
+a lower path limit than was previously received. MAX_CLIENT_PATHS and MAX_SERVER_PATHS frames that
 do not increase the path limit MUST be ignored.
 
-An endpoint MUST NOT initiate a path with a path ID higher than the Maximum Paths value.
-An endpoint MUST terminate the a connection with an error of type MP_PROTOCOL_VIOLATION if a peer opens more paths than was permitted. 
+An endpoint MUST NOT initiate publish MP_NEW_CONNECTION_ID frames with a path ID higher
+than the Maximum Paths value advertised in MAX_CLIENT_PATHS for even numbered paths
+or in MAX_SERVER_PATHS for odd numbers.
```

Comment:
See discussion above. we need to agree on principles before fixing the text.

Reply:
I don't like the "cumulative number of paths" because it is ambiguous -- there are many cases in which the apparent number of paths is not the same at both ends, e.g., if probes are in progress or if abandon is in transit. "MAX_STREAMS" get around that by forcing an automatic opening of streams -- for example, if the only stream is stream 0 and the server receives something for stream 16, then streams 4, 8 and 12 are automatically opened.